UPGRADE-3.4.md

```diff
@@ -277,6 +277,11 @@ Security
    `DigestAuthenticationListener` and `DigestAuthenticationEntryPoint` will be 
    removed in 4.0. Use another authentication system like `http_basic` instead.
 
+ * Using the `AdvancedUserInterface` is now deprecated. To use the existing
+   functionality, create a custom user-checker based on the
+   `Symfony\Component\Security\Core\UserChecker`. This functionality will be
```

Comment:
`Symfony\Component\Security\Core\User\UserChecker`

Reply:
This interface will [...]

Reply:
`UserChecker` !== `UserCheckerInterface`

Shouldnt we deprecate `UserChecker`? as it's tied to `AdvancedUserInterface` :thinking:

Reply:
No, but imo it should be renamed to the `InMemoryUserChecker` (see other comments), just like the `InMemoryUser`. I wish to do this, but I don't have the time to do so before the feature freeze after this gets merged.

If you have time to do so, I'd gladly help out. I'm already stretching my time with non-project related stuff at the moment, so I can't handle another full PR within a few days

Reply:
I see. But in changelogs shouldnt we promote "based on UserCheckerInterface" instead?

Ill have a look at the InMemory story soonish :) i like it.

Reply:
@xabbuh This particular reference (and the one for `CHANGELOG.md`) are correctly referring to the implementation, how the conditions work based on the advanced user interface.